lib/audio_net.py

```diff
@@ -94,23 +100,20 @@ def __init__(self, dataset, net_count = 1, audio_settings = None):
         
         split = int(np.floor(self.validation_split * self.dataset_size))
 
+        # Split the dataset into validation and training data loaders
+        dataset_rng = torch.Generator().manual_seed(self.data_seed)
+        train_dataset, val_dataset = torch.utils.data.random_split(dataset, [self.dataset_size - split, split], generator=dataset_rng)
+
+        np.random.seed(self.random_seed)
+
         for i in range(self.net_count):
             self.nets.append(TinyAudioNet(self.input_size, len(self.dataset_labels), True))
-            self.optimizers.append(optim.SGD(self.nets[i].parameters(), lr=0.003, momentum=0.9, nesterov=True))
-            self.random_seeds.append(random.randint(0, 100000))
-            
- 
-            # Split the dataset into validation and training data loaders
-            indices = list(range(self.dataset_size))
-            np.random.seed(self.random_seeds[i])
-            np.random.shuffle(indices)
-            train_indices, val_indices = indices[split:], indices[:split]
-            self.train_indices.append( train_indices)
-            
-            train_sampler = SubsetRandomSampler(self.train_indices[i])
-            valid_sampler = SubsetRandomSampler(val_indices)
-            self.train_loaders.append(torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, sampler=train_sampler))
-            self.validation_loaders.append(torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, sampler=valid_sampler))
+            optimizer = optim.SGD(self.nets[i].parameters(), lr=0.003, momentum=0.9, nesterov=True)
+            self.optimizers.append(optimizer)
+            self.schedulers.append( torch.optim.lr_scheduler.ReduceLROnPlateau(optimizer, 'min'))
+
+            self.train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=self.batch_size, shuffle=True)
```

Comment:
We create a shuffling data loader, so we don't need to do shuffling during training; that happens for us automatically